spec/outputs/elasticsearch_spec.rb

```diff
@@ -73,6 +73,89 @@
     end
   end
 
+  describe "node client create actions", :elasticsearch => true do
+    require "logstash/outputs/elasticsearch"
+    require "elasticsearch"
+    let(:es) { Elasticsearch::Client.new }
+
+    def get_es_output(action, id = nil)
+      settings = {
+        "manage_template" => true,
+        "index" => "logstash-create",
+        "template_overwrite" => true,
+        "protocol" => "node",
+        "host" => "localhost",
+        "action" => action
+      }
+      settings['document_id'] = id unless id.nil?
+      LogStash::Outputs::ElasticSearch.new(settings)
+    end
+
+    before :each do
+      # Delete all templates first.
+      # Clean ES of data before we start.
+      es.indices.delete_template(:name => "*")
+      # This can fail if there are no indexes, ignore failure.
+      es.indices.delete(:index => "*") rescue nil
+    end
+
+    context "when action => create" do
+      it "should create new documents with or without id" do
+        subject = get_es_output("create", "id123")
+        subject.register
+        subject.receive(LogStash::Event.new("message" => "sample message here"))
+        subject.buffer_flush(:final => true)
+        es.indices.refresh
+        # Wait or fail until everything's indexed.
+        Stud::try(3.times) do
+          r = es.search
+          insist { r["hits"]["total"] } == 1
+        end
+      end
+
+      it "should creat new documents without id" do
+        subject = get_es_output("create")
+        subject.register
+        subject.receive(LogStash::Event.new("message" => "sample message here"))
+        subject.buffer_flush(:final => true)
+        es.indices.refresh
+        # Wait or fail until everything's indexed.
+        Stud::try(3.times) do
+          r = es.search
+          insist { r["hits"]["total"] } == 1
+        end
+      end
+    end
+
+    context "when action => create_unless_exists" do
```

Comment:
can we add a test where a) we populate ES with a doc with the same ID and then b) try to create_unless_exists? that should fail